app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -9,71 +12,109 @@ class OmniauthCallbacksController < Devise::OmniauthCallbacksController
     # SAML doesn't support the CSRF protection
     protect_from_forgery except: %i[mocksaml bird nbp]
 
-    def sso_callback # rubocop:disable Metrics/AbcSize
-      # Check if an existing user is already signed in (passed through the RelayState)
-      # and trying to add a new identity to their account. If so, we load the user information
-      # and set it as the current user. This is necessary to avoid creating a new user.
-      current_user = User.find_by(id: OmniAuth::NonceStore.pop(params[:RelayState]))
-
-      # For existing users, we want to redirect to their profile page after adding a new identity
-      store_location_for(:user, edit_user_registration_path) if current_user.present?
-
-      # The instance variable `@user` is used by Devise internally and should be set here
-      @user = User.from_omniauth(request.env['omniauth.auth'], current_user)
-
-      if @user.persisted?
-        # The `sign_in_and_redirect` will only proceed with the login if the account has been confirmed
-        # either through trust by the IdP or with a confirmation mail. Until then, a flash message
-        # notifies users about the missing email confirmation
-        sign_in_and_redirect @user, event: :authentication
-        # We use the configured OmniAuth camilization to include the user-facing name of the provider in the flash message
-        set_flash_message(:notice, :success, kind: OmniAuth::Utils.camelize(provider)) if is_navigational_format?
```

Comment:
Previously, this flash message was shown on every (successful) sign-in (not only when creating a new account) -- similar to `devise.sessions.signed_in` for password-based logins.

With the current implementation, the message is only shown for successful user registrations (except for the `nbp` provider). Ideally, I would say we should this message every time someone registers for CodeHarbor (except `nbp`) or signs in successfully (any provider) when no other flash message is displayed (i.e., add it to the `sign_in_with_identity`; that should do the job). Please check the error case as well.